### PR TITLE
Led slots API add

### PR DESCRIPTION
**Pull request by tasleson**
Exposes the ledmon slots API.

* Hopefully fixes all known CI issues (except megaraid)

### CODE REVIEW

**Comment by tasleson** [comment]
@trgill @vojtechtrefny Please take a peek at this PR, focus on public API introduced with 5bf397a879af0a509f3a79c373a0f4e35645c5da

File: c_binding/include/libstoragemgmt/libstoragemgmt_local_disk.h [hunk 1]
```diff
@@ -547,6 +547,266 @@ int LSM_DLL_EXPORT lsm_local_disk_link_speed_get(const char *disk_path,
...
+ *      1.10
+ *
+ * Description:
+ *    Return a handle to be used for LED slots API addition.
+ *    This handle is used by most of the API slots methods.
+ *    Make sure to call lsm_led_handle_free() to release resources.
+ *
+ * @handle:
+ *      lsm_led_handle.  Opaque handle pointer to be initialized by this
+ * function.
```
Comment (trgill) [inline c_binding/include/libstoragemgmt/libstoragemgmt_local_disk.h:562]: indent the word "function"?

File: c_binding/lsm_local_disk.cpp [hunk 2]
```diff
@@ -1428,3 +1430,242 @@ int lsm_local_disk_link_speed_get(const char *disk_path, uint32_t *link_speed,
...
+
+const char LSM_DLL_EXPORT *lsm_led_slot_device(lsm_led_slot *slot) {
+    if (LSM_IS_LED_SLOT_ENTRY(slot)) {
+        return led_slot_device(slot->slot);
+    }
+    return NULL;
+}
+
+uint32_t lsm_led_slot_status_get(lsm_led_slot *slot) {
+    uint32_t translated = LSM_DISK_LED_STATUS_UNKNOWN;
```
Comment (trgill) [inline c_binding/lsm_local_disk.cpp:1588]: would it make sense to call !LSM_IS_LED_SLOT_ENTRY(slot) here and return LSM_ERR_INVALID_ARGUMENT?

File: c_binding/include/libstoragemgmt/libstoragemgmt_local_disk.h [hunk 3]
```diff
@@ -547,6 +547,266 @@ int LSM_DLL_EXPORT lsm_local_disk_link_speed_get(const char *disk_path,
...
+ * Return:
+ *      Error code as enumerated by 'lsm_error_number':
+ *          * LSM_ERR_OK
+ *              On success.
+ *          * LSM_ERR_INVALID_ARGUMENT
+ *              When any argument is NULL
+ *          * LSM_ERR_LIB_BUG
+ *              When something unexpected happens.
+ *          * LSM_ERR_PERMISSION_DENIED
+ *              Insufficient permission to initialize handle
```
[the remark this answers is not in the document]
Comment (tasleson) [reply c_binding/include/libstoragemgmt/libstoragemgmt_local_disk.h:575]: @trgill I forgot about these comments.  Looking at them now I see you're correct.  I need to investigate the permissions stuff.  Typically you don't want libraries checking for permissions, just bubble up errors which are caused by insufficient permissions.

**Pull request merged**